src/process_management/process.rs

```diff
@@ -1,43 +1,99 @@
+use crate::{
+    core::socket_handler::SocketHandler,
+    api::schema::instructions::{DeserializableCoreInstr, SerializablePluginInstr}
+};
+
 use std::{
     process::{Child, Command, Stdio},
-    ffi::OsStr
+    fmt::Debug, path::PathBuf,
+    sync::Arc, time::Duration,
+    thread::sleep
 };
+use log::{warn, debug, error, trace};
 
-use log::{warn, debug};
 use anyhow::Result;
+use serde::Serialize;
+use tokio::{task::JoinHandle, sync::{ Mutex, mpsc::{self, Receiver, Sender}}, time::timeout};
+
 #[derive(Debug)]
 pub struct Process {
     child: Child,
+    process_path: PathBuf,
+    core_read_thread: JoinHandle<()>,
+    socket: Arc<Mutex<SocketHandler>>,
```

Comment:
Do you think we'll be able to lock it separately in the future to allow concurrent read and writes? That should work with IPC.

Reply:
Would require some work on SocketHandler..  Exactly how much work that would entail is unknown, let's open an issue for this and then revisit in a different PR.

Reply:
Issue created.